src/plugins/chapters/index.js

```diff
@@ -0,0 +1,195 @@
+        );
+        textTrack.addCue(cue);
+      });
+      this.chaptersTrack = textTrack;
+      this.setupProgressBarMarkers();
+      this.setupProgressBarChapter();
+      this.setupControlBarChapter();
+    }
+  };
+
+  /**
+   * Setup the controlbar chapter display.
+   */
+  ChaptersPlugin.prototype.setupControlBarChapter = function setupControlBarChapter() {
+    const controlBarChapterHolder =
+      this.player.$('.vjs-control-bar-chapter-display') || document.createElement('div');
+    controlBarChapterHolder.setAttribute('class', 'vjs-control-bar-chapter-display');
+
+    const wrapper = this.player.$('.vjs-control-bar .vjs-spacer');
+    wrapper.innerHTML = '';
+    wrapper.classList.add('vjs-control-bar-chapter-wrapper');
+    wrapper.appendChild(controlBarChapterHolder);
+
+    this.chaptersTrack.addEventListener('cuechange', () => {
+      controlBarChapterHolder.innerHTML =
+        this.chaptersTrack.activeCues_.length > 0 ? this.chaptersTrack.activeCues_[0].text : '';
+    });
+  };
+
+  /**
+   * Setup the progress bar markers.
+   */
+  ChaptersPlugin.prototype.setupProgressBarMarkers = function setupProgressBarMarkers() {
+    const total = this.player.duration();
+    const { seekBar } = this.player.controlBar.progressControl;
+    this.chaptersTrack.cues_.forEach(marker => {
+      if (marker.startTime !== 0) {
+        const markerTime = marker.startTime;
+        const left = (markerTime / total) * 100 + '%';
+
+        const markerEl = videojs.dom.createEl('div', undefined, {
+          class: 'vjs-chapter-marker',
+          style: `left: ${left}`
+        });
+
+        seekBar.el().append(markerEl);
+      }
+    });
+  };
+
+  /**
+   * Setup the progrees bar on-hover chapter display.
+   */
+  ChaptersPlugin.prototype.setupProgressBarChapter = function setupProgressBarChapter() {
+    const chapterEl = videojs.dom.createEl('div', undefined, {
+      class: 'vjs-chapter-display',
+      style: `width: ${this.player.$('.vjs-vtt-thumbnail-display').style.width || '160px'}`
+    });
+
+    const timeTooltip = this.player.getDescendant([
+      'controlBar',
+      'progressControl',
+      'seekBar',
+      'mouseTimeDisplay',
+      'timeTooltip'
+    ]);
+
+    timeTooltip.el().parentElement.prepend(chapterEl);
+
+    const getChapterFromPoint = point => {
+      const total = this.player.duration();
+      const seekBarTime = point * total;
+      const chapter = this.chaptersTrack.cues_.find(marker => {
+        return marker.startTime <= seekBarTime && marker.endTime >= seekBarTime;
+      });
+      return chapter ? chapter.text : '';
+    };
+
+    timeTooltip.update = function (seekBarRect, seekBarPoint, content) {
+      const originalUpdateFn = Object.getPrototypeOf(this).update;
+      originalUpdateFn.call(this, seekBarRect, seekBarPoint, content);
+      chapterEl.innerHTML = getChapterFromPoint(seekBarPoint);
+    };
+  };
+
+  return ChaptersPlugin;
+})();
+
+export default chapters;
```

Comment:
I understood that the plugin code is probably a template that gets filled in, it is worth checking if we could have a more modern template (Using class instead of prototype and using arrow functions instead of regular functions) .

Reply:
Yeah, not really a template, it all started with the seek-thumbnails plugin that was copied as-is from somewhere.
I realize it's not modern, but it's actually pretty clear, readable and well structured which is what I find most important.
Switching to a Class would be easy, but if I would re-write these I'd go the functional route, getting rid of all these `this` :)